src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1114,12 +1114,16 @@ private void resolveToken(ref CORINFO_RESOLVED_TOKEN pResolvedToken)
             pResolvedToken.hClass = null;
             pResolvedToken.hMethod = null;
             pResolvedToken.hField = null;
-
+            
             if (result is MethodDesc)
             {
                 MethodDesc method = result as MethodDesc;
                 pResolvedToken.hMethod = ObjectToHandle(method);
                 pResolvedToken.hClass = ObjectToHandle(method.OwningType);
+
+#if READYTORUN
+            _compilation.NodeFactory.AddModuleTokenForMethod(method, new ModuleToken(_tokenContext, (mdToken)pResolvedToken.token));
```

Comment:
I was hoping we would be stashing this information in a map that is local to `CorInfoImpl` and use the mapping when `CorInfoImpl` needs to talk to an API on `NodeFactory` that needs the token, but we don't have it at that point (like in the delegate construction case that prompted this). Are there other places where this would be insufficient? I'm worried that storing this in a global place like the `NodeFactory` doesn't actually solve the problem of us being able to reference the same method through different tokens, and which token to use when.

This is an improvement in the sense that we don't have to reverse scan the ECMA metadata table, but it doesn't go as far as I hoped it would.